sdk/trace/src/test/scala/org/typelevel/otel4s/sdk/trace/SdkTracerSuite.scala

```diff
@@ -0,0 +1,1004 @@
+/*
+ * Copyright 2023 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.sdk.trace
+
+import cats.effect.IO
+import cats.effect.IOLocal
+import cats.effect.MonadCancelThrow
+import cats.effect.Resource
+import cats.effect.std.Random
+import cats.effect.testkit.TestControl
+import cats.syntax.functor._
+import cats.~>
+import munit.CatsEffectSuite
+import munit.Location
+import munit.TestOptions
+import org.typelevel.otel4s.Attribute
+import org.typelevel.otel4s.Attributes
+import org.typelevel.otel4s.context.propagation.PassThroughPropagator
+import org.typelevel.otel4s.context.propagation.TextMapPropagator
+import org.typelevel.otel4s.sdk.common.InstrumentationScope
+import org.typelevel.otel4s.sdk.context.Context
+import org.typelevel.otel4s.sdk.trace.data.EventData
+import org.typelevel.otel4s.sdk.trace.data.SpanData
+import org.typelevel.otel4s.sdk.trace.data.StatusData
+import org.typelevel.otel4s.sdk.trace.exporter.InMemorySpanExporter
+import org.typelevel.otel4s.sdk.trace.processor.SimpleSpanProcessor
+import org.typelevel.otel4s.sdk.trace.processor.SpanProcessor
+import org.typelevel.otel4s.sdk.trace.samplers.Sampler
+import org.typelevel.otel4s.trace.Span
+import org.typelevel.otel4s.trace.Status
+import org.typelevel.otel4s.trace.Tracer
+import org.typelevel.otel4s.trace.TracerProvider
+
+import scala.concurrent.duration._
+import scala.util.control.NoStackTrace
+
+class SdkTracerSuite extends CatsEffectSuite {
```

Comment:
Copy-paste from the `oteljava` module.

I think the proper solution should be:
1) Make an abstract suite in the `testkit-trace` tests:
```scala
abstract class SdkTracerSharedSuite extends CatsEffectSuite {
  def makeSdk(
      sampler: Sampler,
      additionalPropagators: List[TextMapPropagator[Context]]
  ): Resource[IO, SdkTracerSharedSuite.Sdk]
}
```

2) Both `sdk` and `oteljava` extend their respective SdkTracerSuite from the abstract one. 

3) We are sure both implementation run against the same test suite